cypress/integration/networking.spec.ts

```diff
@@ -3,49 +3,165 @@ import * as path from "path";
+          commands.pick("peer-selector").contains("rudolfs").should("exist");
+          commands.pick("peer-selector").contains("maintainer").should("exist");
+          commands.pick("peer-selector").click();
+
+          cy.log("the current user does not show up in the peer selector");
+          commands
+            .pick("peer-dropdown-container")
+            .contains("abbey")
+            .should("not.exist");
+
+          cy.log("add contributor remote on maintainer's node");
+          commands.pick("sidebar", "profile").click();
+          commands.pick("peer-id").then(el => {
+            const peerId = el.attr("title");
+            if (!peerId) {
+              throw new Error("Could not find PeerId");
+            }
+            const shortenedPeerId = peerId.slice(0, 7);
+
+            nodeManager.asNode(maintainerNode);
+            commands.pick("project-list-entry-new-fancy-project.xyz").click();
+
+            commands.pick("peer-selector").click();
+            commands.pick("manage-remotes").click();
+
+            commands.pick("followed-peers", "peer-abbey").should("not.exist");
+            cy.contains("Still looking").should("not.exist");
+
+            commands.pick("peer-input").type(peerId);
+            commands.pick("follow-button").click();
+
+            cy.log("the remote shows up in the waiting area");
+            cy.contains("Still looking").should("exist");
+            commands
+              .pick("pending-peers")
+              .contains(shortenedPeerId)
+              .should("exist");
+
+            cy.log("the remote moves out of the waiting area");
+            commands
+              .pick("followed-peers", "peer-abbey")
+              .contains(shortenedPeerId)
+              .should("exist");
+            cy.contains("Still looking").should("not.exist");
+            commands
+              .pick("followed-peers", "peer-abbey", "follow-toggle")
+              .contains("Following")
+              .should("exist");
+          });
+
+          cy.log("add a random non-existing remote");
+          const nonExistingRemote =
+            "hydysjnrpb4gwtsz66ei9xsyzta1afuhc1h6neufmfghozduo4p3ua";
+          const shortenedNonExistingRemote = nonExistingRemote.substr(0, 7);
+
+          cy.contains("Still looking").should("not.exist");
+
+          commands.pick("peer-input").type(nonExistingRemote);
+          commands.pick("follow-button").click();
+
+          cy.log("check that remote stays in waiting area");
+          cy.wait(1000);
+          cy.contains("Still looking").should("exist");
+          commands.pick("pending-peers").contains(shortenedNonExistingRemote);
+
+          cy.log("remove the pending remote");
+          commands.pick("pending-peers", "follow-toggle").click();
+          cy.contains("Still looking").should("not.exist");
+          commands.pick("pending-peers").should("not.exist");
+        },
+        { node1Handle: "rudolfs", node2Handle: "abbey" }
```

Comment:
I don’t think it’s ideal to have the parameters here at the very end, disconnected from `withTwoOnboardedNodes()`. I propose we address this be replacing the callback pattern with the more common promise chains. That is
```
startTwoOnborardedNodes({ node1Handle: "rudolfs", node2Handle: "abbey" }).then(([node1, node2]) => {
  // ...
})
```

Reply:
I'm not sure how to do this, because I want to encapsulate the cleanup in the callback wrapper. One way to do it would be using the disposer pattern, but then we still would have to pass around a callback. Another way would be to just flip the arguments around, i.e. options first, then callback. The drawback with this approach would be losing option defaults.

Happy to pair on this if you have a clear idea on how to make this better.

Reply:
> I'm not sure how to do this, because I want to encapsulate the cleanup in the callback wrapper.

You’re right. I didn’t realize that the cleanup is now part of `startTwoOnborardedNodes`.


> Another way would be to just flip the arguments around, i.e. options first, then callback. The drawback with this approach would be losing option defaults.

I’d prefer this but will defer to you. We can still have some optionality by making the fields optional and doing something like `startTwoOnboardedNodes({}, () => { ... })`.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1559/commits/e32842d80a5401300abd3a5b71882c3a83197f21